lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -271,6 +283,22 @@ def write(chunk)
             entry['metadata']['severity'] = 'DEFAULT'
           end
 
+          # If a field is present in the label_map, send its value as a label
+          # (mapping the field name to label name as specified in the config)
+          # and do not send that field as part of the payload.
+          if !label_map.nil?
```

Comment:
The values are parsed out of each log entry so they could be different for each one.

It's possible the label/value pair could end up being the same for all entries in the batch, I could write a loop later to detect that although I'm not sure if it's worth the effort.